File: SPRING/Hospital_Management_system_angular/src/app/services/dietary/meal-schedule.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { MealSchedule } from '../../models/dietary/meal-schedule.model';

@Injectable({ providedIn: 'root' })
export class MealScheduleService {
  private api = environment.apiUrl + 'meal-schedules';

  constructor(private http: HttpClient) {}

  getAll(): Observable<MealSchedule[]> {
    return this.http.get<MealSchedule[]>(this.api);
  }
  getById(id: number): Observable<MealSchedule> {
    return this.http.get<MealSchedule>(`${this.api}/${id}`);
  }
  getActive(): Observable<MealSchedule[]> {
    return this.http.get<MealSchedule[]>(`${this.api}/active`);
  }
  getCurrentMeal(): Observable<MealSchedule> {
    return this.http.get<MealSchedule>(`${this.api}/current`);
  }
  create(data: MealSchedule): Observable<MealSchedule> {
    return this.http.post<MealSchedule>(this.api, data);
  }
  update(id: number, data: MealSchedule): Observable<MealSchedule> {
    return this.http.put<MealSchedule>(`${this.api}/${id}`, data);
  }
  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.api}/${id}`);
  }
}

File: SPRING_BOOT/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/MealScheduleService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.entity.MealSchedule;
import java.util.List;

public interface MealScheduleService {
    MealSchedule create(MealSchedule schedule);
    MealSchedule getById(Long id);
    List<MealSchedule> getAll();
    List<MealSchedule> getActive();
    MealSchedule update(Long id, MealSchedule schedule);
    void delete(Long id);
    MealSchedule getCurrentMeal();
}
